### PR TITLE
Add distinct type support to object type defn

### PR DESCRIPTION
**Pull request by rdhananjaya**
$title 

Fixes https://github.com/ballerina-platform/ballerina-lang/issues/26110

## Check List 
- [ ] Read the [Contributing Guide](https://github.com/ballerina-platform/ballerina-lang/blob/master/CONTRIBUTING.md)
- [ ] Updated Change Log
- [ ] Checked Tooling Support (#<Issue Number>)
- [ ] Added necessary tests
  - [ ] Unit Tests
  - [ ] Spec Conformance Tests
  - [ ] Integration Tests
  - [ ] Ballerina By Example Tests
- [ ] Increased Test Coverage   
- [ ] Added necessary documentation  
  - [ ] API documentation 
  - [ ] Module documentation in Module.md files
  - [ ] Ballerina By Examples

### CODE REVIEW

File: compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/parser/BLangNodeTransformer.java [hunk 1]
```diff
@@ -5030,6 +5030,9 @@ private boolean isPresent(Node node) {
 
     private boolean checkIfAnonymous(Node node) {
         Node parent = node.parent();
+        if (parent.kind() == SyntaxKind.DISTINCT_TYPE_DESC) {
```
Comment (nadeeshaan) [inline compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/parser/BLangNodeTransformer.java:5033]: Can we collapse this if statement with the next line's return statement?

File: compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/model/types/BTypeIdSet.java [hunk 2]
```diff
@@ -43,7 +43,12 @@ public BTypeIdSet(Set<BTypeId> primary, Set<BTypeId> secondary) {
 
     public BTypeIdSet(Set<BTypeId> primary) {
         this.primary = primary;
-        this.secondary = emptySet;
+        this.secondary = new HashSet<>();
+    }
+
+    public BTypeIdSet() {
```
Comment (nadeeshaan) [inline compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/model/types/BTypeIdSet.java:49]: Shall we chain this constructor to the one with the most arguments (line 44)?

Comment (rdhananjaya) [reply compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/model/types/BTypeIdSet.java:48]: fixed

File: compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/parser/BLangNodeTransformer.java (hunk 1 above)
Comment (rdhananjaya) [reply compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/parser/BLangNodeTransformer.java]: fixed

File: tests/jballerina-unit-test/src/test/java/org/ballerinalang/test/object/DistinctObjectTest.java [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+import org.testng.annotations.Test;
+
+import static org.ballerinalang.test.util.BAssertUtil.validateError;
+
+/**
+ * Test cases for distinct object type definitions.
+ *
+ * @since 2.0
+ */
+public class DistinctObjectTest {
```
Comment (hasithaa) [inline tests/jballerina-unit-test/src/test/java/org/ballerinalang/test/object/DistinctObjectTest.java:32]: Shall we add runtime test cases as well?

Comment (rdhananjaya) [reply tests/jballerina-unit-test/src/test/java/org/ballerinalang/test/object/DistinctObjectTest.java:32]: Added in https://github.com/ballerina-platform/ballerina-lang/pull/26098/commits/a7e2849c8dc4a7b749f591b59979b504a6612168

**Review by KRVPerera — APPROVED** [review]

**Review by hasithaa — APPROVED** [review]

**Pull request merged**